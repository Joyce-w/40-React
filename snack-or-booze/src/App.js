import React, { useState, useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import "./App.css";
import Home from "./Home";
import SnackOrBoozeApi from "./Api";
import NavBar from "./NavBar";
import { Route, Switch } from "react-router-dom";
import Menu from "./FoodMenu";
import Snack from "./FoodItem";
import NewItemForm from "./NewItemForm";

function App() {

  //how to update snacks/drinks if it is updated in another server?
  const [isLoading, setIsLoading] = useState([]);
  const [snacks, setSnacks] = useState([]);
  const [drinks, setDrinks] = useState([]);

  //update loading
  const updateLoading = () => {
    setIsLoading((isLoading) => [...isLoading, false]);
  }

  //get snack data
  useEffect(() => {
    async function getSnacks() {
      let snacks = await SnackOrBoozeApi.getSnacks();
      setSnacks(snacks);
      updateLoading();
    }
    getSnacks();
  }, []);

// get drink data
  useEffect(() => {
    async function getDrinks() {
      let drinks = await SnackOrBoozeApi.getDrinks();
      setDrinks(drinks);
      updateLoading();
    }
    getDrinks();
  }, []);

  let loading = isLoading.length !== 2 ? <p>Loading &hellip;</p> : null;


  return (
    <div className="App">
      <BrowserRouter>
        <NavBar  />
        <main>
          <Switch>
            <Route exact path="/">
              <Home loading={loading} snacks={snacks.length} drinks={drinks.length} />
            </Route>
            {/* Snacks Route */}
            <Route exact path="/snacks">
              <Menu items={snacks} title="Snacks" category="Snacks"/>
            </Route>
            <Route path="/snacks/:id">
              <Snack items={snacks} cantFind="/snacks" />
              
              {/* Drinks route */}
            </Route>
            <Route exact path="/drinks">
              <Menu items={drinks} title="Drinks" category="Drinks"/>
            </Route>
            <Route path="/drinks/:id">
              <Snack items={drinks} cantFind="/drinks" />
            </Route>
            {/* New Item Form */}
            <Route exact path="/newItem">
              <NewItemForm/>
            </Route>
            <Route>
              <p>Hmmm. I can't seem to find what you want.</p>
            </Route>
          </Switch>
        </main>
      </BrowserRouter>
    </div>
  );
}

export default App;
