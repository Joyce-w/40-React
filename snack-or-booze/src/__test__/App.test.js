import React from 'react'
import { render } from '@testing-library/react'
import App from '../App'

//smoke test 
it("should render w/o crashing", () => {
    render(<App />)
})

//snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<App />)
    expect(asFragment()).toMatchSnapshot();
})

//specialized tests
// it("should have text when rendered", function () {
//     const { getByText } = render(<App />)
//     let p = getByText('Welcome', {exact: false})
//     expect(p).toBeInTheDocument();
// })

