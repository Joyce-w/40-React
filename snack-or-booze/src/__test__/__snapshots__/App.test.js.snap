// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <nav
        class="navbar navbar-expand-md"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          Snack or Booze
        </a>
        <ul
          class="ml-auto navbar-nav"
        >
          <a
            href="/snacks"
          >
            Snacks
          </a>
          <a
            href="/drinks"
          >
            Drinks
          </a>
          <a
            href="/newItem"
          >
            New Item
          </a>
        </ul>
      </nav>
    </div>
    <main>
      <section
        class="col-md-8"
      >
        <div
          class="card"
        >
          <div
            class="text-center card-body"
          >
            <h5
              class="card-title"
            />
            <h3
              class="font-weight-bold"
            >
              Welcome to Silicon Valley's premier dive cafe!
            </h3>
            <p>
              Loading â€¦
            </p>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
